internal/app/osd/internal/reg/reg.go

```diff
@@ -271,57 +272,33 @@ func (r *Registrator) Dmesg(ctx context.Context, in *empty.Empty) (data *proto.D
 // be requested and the contents of the log file are streamed in chunks.
 // nolint: gocyclo
 func (r *Registrator) Logs(req *proto.LogsRequest, l proto.OSD_LogsServer) (err error) {
-	var (
-		client *containerd.Client
-		ctx    context.Context
-		pods   []*pod
-		task   *tasks.GetResponse
-	)
-
-	pods, err = podInfo(req.Namespace)
-	if err != nil {
-		return err
+	var filename string
+
+	switch {
+	case req.Namespace == "system" || req.Id == "kubelet" || req.Id == "kubeadm":
```

Comment:
should we do same or close to that for all the services?

i.e. if service is in `system.Services`, we can look up by id, ask service for the log file name and use it from there?

(that was my thought)